lib/src/simple_token.dart

```diff
@@ -0,0 +1,137 @@
+
+library angular_ast.src.simple_token;
+
+import 'package:quiver/core.dart';
+
+part 'token/simple_type.dart';
+
+class NgSimpleToken {
+  factory NgSimpleToken.bang(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.bang, offset);
+  }
+
+  //Probably don't need
+  factory NgSimpleToken.closeBrace(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeBrace, offset);
+  }
+
+  factory NgSimpleToken.closeBracket(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeBracket, offset);
+  }
+
+  factory NgSimpleToken.closeParen(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.closeParen, offset);
+  }
+
+  factory NgSimpleToken.commentBegin(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.commentBegin, offset);
+  }
+
+  factory NgSimpleToken.commentEnd(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.commentEnd, offset);
+  }
+
+  factory NgSimpleToken.dash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.dash, offset);
+  }
+
+  factory NgSimpleToken.doubleQuote(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.doubleQuote, offset);
+  }
+
+  factory NgSimpleToken.doubleQuotedText(int offset, String lexeme) {
+    return new NgSimpleToken(
+        new NgSimpleTokenType.doubleQuotedText(lexeme), offset);
+  }
+
+  factory NgSimpleToken.tagStart(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.tagStart, offset);
+  }
+
+  factory NgSimpleToken.tagEnd(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.tagEnd, offset);
+  }
+
+  factory NgSimpleToken.EOF(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.EOF, offset);
+  }
+
+  factory NgSimpleToken.equalSign(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.equalSign, offset);
+  }
+
+  factory NgSimpleToken.forwardSlash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.forwardSlash, offset);
+  }
+
+  factory NgSimpleToken.hash(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.hash, offset);
+  }
+
+  //Probably don't need
+  factory NgSimpleToken.openBrace(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openBrace, offset);
+  }
+
+  factory NgSimpleToken.openBracket(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openBracket, offset);
+  }
+
+  factory NgSimpleToken.openParen(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.openParen, offset);
+  }
+
+  factory NgSimpleToken.singleQuote(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.singleQuote, offset);
+  }
+
+  factory NgSimpleToken.singleQuotedText(int offset, String lexeme) {
+    return new NgSimpleToken(
+        new NgSimpleTokenType.singleQuotedText(lexeme), offset);
+  }
+
+  factory NgSimpleToken.star(int offset) {
+    return new NgSimpleToken._(NgSimpleTokenType.star, offset);
+  }
+
+  factory NgSimpleToken.text(int offset, String lexeme) {
```

Comment:
I wanted to keep this tokenizer to be agnostic about what can be considered a text node block or an identifier and let the parser that utilizes this tokenizer decide on how to treat the text. I've instead created different text regexp for text, comment, and identifier just so text blocks with spaces won't be treated as multiple redundant texts. 

We can further discuss what the best course of action for this is; I feel like if we classify texts as identifier, textnode text, or comment - we wouldn't be deviating much from the initial scanner/lexer implemented.